soroban-react-dapp/contracts/authorized_greeting/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+#![no_std]
+use soroban_sdk::{contract, contractimpl, Env, Symbol, symbol_short, String, Address, Map};
+
+const GREET: Symbol = symbol_short!("GREET");
+const AUTH_GRTRS: Symbol = symbol_short!("AUTH_GTRS");
+const ADMIN: Symbol = symbol_short!("ADMIN");
+
+#[contract]
+pub struct AuthorizedGreetingContract;
+
+#[contractimpl]
+impl AuthorizedGreetingContract {
+    pub fn set_admin(env: Env, admin: Address) {
+        env.storage().instance().set(&ADMIN, &admin);
+    }
+
+    pub fn get_admin(env: &Env) -> Address {
+        env.storage().instance().get(&ADMIN).expect("Admin not set in storage")
+    }
+
+    pub fn add_greeter(env: Env, greeter: Address) {
+        let admin = Self::get_admin(&env);  
+        admin.require_auth();
+
+        let mut authorized_greeter_list = Self::get_authorized_greeter_list(&env);
+        authorized_greeter_list.set(greeter, true);
+        env.storage().instance().set(&AUTH_GRTRS, &authorized_greeter_list);
+    }
+
+    pub fn get_authorized_greeter_list(env: &Env) -> Map<Address, bool> {
```

Comment:
This mapping can be attacked by  DOS.
See https://github.com/paltalabs/instance-persistent-dos-soroban

Reply:
If only admin can add greeters the scale up of the storage is closely limited, so you refer to the GET public function, a faster solution is remove "pub" and make the function private to only the consumition from add_greeter fn but that impacts directly on tests scripts.